File: fibonacci.pl
# Fibonacci performance test
# Author Stoyan Rangelov (devsoul) <stoyan.rangelov@gmail.com>
sub fibonacci {
    my $n = shift;

    if ($n < 2) {
        return $n;
    }
    return fibonacci($n-1) + fibonacci($n-2);
}

print fibonacci(30)."\n";

File: fibonacci.c
/**
 * Fibonacci performance test
 * 
 * @author Stoyan Rangelov (devsoul) <stoyan.rangelov@gmail.com>
 */

#include <stdio.h>

int fibonacci(int n) {
    if (n < 2) {
        return n;
    }
    return fibonacci(n - 2) + fibonacci(n - 1);
}

int main() {
    printf("%d\n", fibonacci(30));
    return 0;
}


